backpack.tf extended sorting.user.js

```diff
@@ -871,6 +874,65 @@ class</a></li>
 
 		}
 
+		var cnSort = 0;
+
+		function sortByCraftNumber() {
+			const sortCN = (a, b) => {
+				try {
+					return Number($(a)[0].outerText.split("#")[1]) - Number($(b)[0].outerText.split("#")[1]);
+				} catch (e) {
+					return parseInt(a[0].split(" ")[1]) - parseInt(b[0].split(" ")[1]);
+				}
+			}
+			let sorts = [
+				(a, b) => sortCN(a,b),
+				(a, b) => sortCN(b,a)
+			]
+			let cn = {};
+			cn["Craft Number"] = {
+				"cc": ["#DD5522"],
+				"items": []
+			};
+
+			cn["No Craft Number"] = {
+				"cc": ["#676780"],
+				"items": []
+			};
+
+			cn["Hidden Items"] = {
+				"cc": ["#676780"]
+			};
+
+			let z = $(".backpack-page .item:not(.spacer)");
+
+			cn["Hidden Items"]["items"] = $(".temp-page .item:not(.spacer)");
+			for (let p = 0; p < z.length; p++) {
+				if ($(z[p])[0].outerText.includes("#")) {
+					cn["Craft Number"]["items"].push($(z[p])[0]);
+				} else {
+					cn["No Craft Number"]["items"].push($(z[p])[0]);
+				}
+
+			}
+
+			for (let k in cn) {
+				if (k == "Craft Number") {
+					cn[k]["items"] = cn[k]["items"].sort(sorts[cnSort]);
+				} else {
+					cn[k]["items"] = genericItemSort("data-price", cn[k]["items"]);
+				}
+			}
+
+			genericSort(cn, "craftnumber", false, {
+				"use": true,
+				"funct": sorts[cnSort]
+			});
```

Comment:
For what reason do you supply your sorting function to `genericSort`?
Generic Sort decides on the order of the categories, your sorting sorts based on numbers, but the categories are strings.

As far as I can see there is no reason to not just use:

```suggestion
			genericSort(cn, "craftnumber", true);
```